src/test/java/htsjdk/samtools/reference/FastaReferenceWriterTest.java

```diff
@@ -195,6 +192,18 @@ public void testAddingInvalidSequenceName(final String invalidName) throws IOExc
         }
     }
 
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void testAddingGziIndexToNonBCFFile() throws IOException {
+        final Path testOutput = File.createTempFile("fwr-test", ".fasta").toPath();
+        final Path testOutputGZI = File.createTempFile("fwr-test", ".fasta.gzi").toPath();
+        IOUtil.deleteOnExit(testOutput);
+        Files.delete(testOutput);
+
+        try (FastaReferenceWriter writer = new FastaReferenceWriterBuilder().setFastaFile(testOutput).setMakeFaiOutput(false).setMakeDictOutput(false).setGziIndexFile(testOutputGZI).build()) {
+            writer.startSequence(testOutput.toString());
+        }
```

Comment:
The `writer.startSequence` call is unnecessary/redundant.

Reply:
fair point